sdks/python/apache_beam/io/iobase.py

```diff
@@ -847,12 +848,21 @@ def expand(self, pbegin):
 
     debug_options = self.pipeline._options.view_as(DebugOptions)
     if debug_options.experiments and 'beam_fn_api' in debug_options.experiments:
-      NUM_SPLITS = 1000
       source = self.source
+
+      def split_source(unused_impulse):
+        total_size = source.estimate_size()
+        if total_size:
+          # 1MB = 1 shard, 1GB = 32 shards, 1TB = 1000 shards, 1PB = 32k shards
+          chunk_size = max(1 << 20, 1000 * int(math.sqrt(total_size)))
```

Comment:
Should we use default of 64MB here instead of 1MB.
total_size is in bytes so why are we multiplying with 1000?

Reply:
I mean 1mb chunk size is only used on files under 5mb which seems pretty reasonable?

Reply:
We want smaller-than-normal chunks when we know the input size is small.